lib/connextclient/ConnextClient.ts

```diff
@@ -260,9 +268,34 @@ class ConnextClient extends SwapClient {
     return this._totalOutboundAmount.get(currency) || 0;
   }
 
-  public maxChannelInboundAmount = (_currency: string): number => {
-    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
-    return MAX_AMOUNT;
+  public checkInboundCapacity = (inboundAmount: number, currency: string) => {
+    const inboundCapacity = this._maxChannelInboundAmount.get(currency) || 0;
+    if (inboundCapacity < inboundAmount) {
+      // we do not have enough inbound capacity to receive the specified inbound amount so we must request collateral
+      this.logger.debug(`collateral of ${inboundCapacity} for ${currency} is insufficient for order amount ${inboundAmount}`);
+
+      // we want to make a request for the current collateral plus the greater of any
+      // minimum request size for the currency or the capacity shortage + 5% buffer
+      const quantityToRequest = inboundCapacity + Math.max(
+        inboundAmount * 1.05 - inboundCapacity,
+        ConnextClient.MIN_COLLATERAL_REQUEST_SIZES[currency] ?? 0,
+      );
+      const unitsToRequest = this.unitConverter.amountToUnits({ currency, amount: quantityToRequest });
+
+      // first check whether we already have a pending collateral request for this currency
+      // if not start a new request, and when it completes call channelBalance to refresh our inbound capacity
+      const requestCollateralPromise = this.requestCollateralPromises.get(currency) ?? this.sendRequest('/request-collateral', 'POST', {
+        assetId: this.tokenAddresses.get(currency),
+        amount: unitsToRequest.toLocaleString('fullwide', { useGrouping: false }),
+      }).then(() => {
+        this.logger.debug(`completed collateral request of ${unitsToRequest} ${currency} units`);
+        this.requestCollateralPromises.delete(currency);
+        return this.channelBalance(currency);
```

Comment:
I think so if we want the `catch` statement to catch any errors thrown by the `channelBalance` method. Otherwise the promise that method returns isn't tracked by anything.